modules/version.nix

```diff
@@ -0,0 +1,79 @@
+{ lib, pkgs, options, config, ... }:
+with builtins; with lib;
+{
+
+  options = with types; {
+    wsl.version =
+      let
+        versionFile = splitString "\n" (readFile ../VERSION);
+      in
+      {
+        release = mkOption {
+          description = "the NixOS-WSL release";
+          type = str;
+          default = elemAt versionFile 0;
+        };
+        rev = mkOption {
+          description = "the NixOS-WSL git revision";
+          type = str;
+          default = elemAt versionFile 1;
+        };
+        systemd = mkOption {
+          type = enum [ "native" "syschdemd" ];
+          description = "the systemd implementation used by NixOS-WSL";
+          default = if config.wsl.nativeSystemd then "native" else "syschdemd";
+        };
+      };
```

Comment:
Those should be internal.

Reply:
How'd you suggest doing that? I think the values should ideally stay accessible as read-only. `rev` is currently being set in flake.nix

Reply:
I mean this option https://github.com/NixOS/nixpkgs/blob/cc8def18f0afbc28fbdd26d2f5972dc6cb158b7e/nixos/lib/testing/network.nix#L74